x-pack/solutions/search/plugins/enterprise_search/server/deprecations/index.ts

```diff
@@ -0,0 +1,366 @@
+            body: {
+              ids: nativeConnectors.map((it) => it.id),
+            },
+          },
+        },
+      });
+    } else {
+      const nativeServiceTypes = [
+        'azure_blob_storage',
+        'box',
+        'confluence',
+        'dropbox',
+        'github',
+        'gmail',
+        'google_cloud_storage',
+        'google_drive',
+        'jira',
+        'mssql',
+        'mongodb',
+        'mysql',
+        'network_drive',
+        'notion',
+        'onedrive',
+        'oracle',
+        'outlook',
+        'postgresql',
+        's3',
+        'salesforce',
+        'servicenow',
+        'sharepoint_online',
+        'sharepoint_server',
+        'slack',
+        'microsoft_teams',
+        'zoom',
+      ];
+      const nativeTypesStr = '[' + nativeServiceTypes.join(', ') + ']';
+      const fauxNativeConnectors = nativeConnectors.filter(
+        (hit) => !nativeServiceTypes.includes(hit.service_type!)
+      );
+
+      if (fauxNativeConnectors.length > 0) {
+        // There are some illegal service_types in their native connectors
+        deprecations.push({
+          level: 'critical',
+          deprecationType: 'feature',
+          title: i18n.translate('xpack.enterpriseSearch.deprecations.fauxNativeConnector.title', {
+            defaultMessage: 'Connectors with `is_native: true` must be of supported service types',
+          }),
+          message: i18n.translate(
+            'xpack.enterpriseSearch.deprecations.fauxNativeConnector.message',
+            {
+              values: { serviceTypes: nativeTypesStr },
+              defaultMessage:
+                'Only specific service types are supported for Elastic-managed connectors. ' +
+                'Eligible service types for Elastic-managed connectors are: {serviceTypes}' +
+                'Connectors of other service types must be converted to Connector Clients before upgrading. ' +
+                'This is a lossless operation, and can be attempted with "quick resolve". ' +
+                'Alternatively, deleting these connectors with mismatched service types will also unblock your upgrade.',
+            }
+          ),
+          documentationUrl:
+            'https://elastic.co/guide/en/enterprise-search/current/upgrading-to-9-x.html',
+          correctiveActions: {
+            manualSteps: [
+              i18n.translate(
+                'xpack.enterpriseSearch.deprecations.fauxNativeConnector.listConnectors',
+                {
+                  defaultMessage: 'Enumerate all connector records',
+                }
+              ),
+              i18n.translate(
+                'xpack.enterpriseSearch.deprecations.fauxNativeConnector.convertPretenders',
+                {
+                  values: { serviceTypes: nativeTypesStr },
+                  defaultMessage:
+                    '"Convert to Client" any that have `is_native: true` but have a `service_type` NOT in: {serviceTypes}.',
```

Comment:
```suggestion
                    'Select "Convert to Client" for any connectors where `is_native: true` but the `service_type` is NOT supported, per the list above.',
```

Because the list is messy and we've already printed it once, I think it's safe to point users up rather than printing it again. LMKWYT.